pkg/utils/fastthrift/fastthrift.go

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2022 CloudWeGo Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fastthrift
+
+import (
+	"errors"
+
+	"github.com/bytedance/gopkg/lang/dirtmake"
+
+	"github.com/cloudwego/kitex/pkg/protocol/bthrift"
+	thrift "github.com/cloudwego/kitex/pkg/protocol/bthrift/apache"
+)
+
+// FastMarshal marshals the msg to buf. The msg should be generated by Kitex tool and implement ThriftMsgFastCodec.
+func FastMarshal(msg bthrift.ThriftMsgFastCodec) []byte {
+	sz := msg.BLength()
+	buf := dirtmake.Bytes(sz, sz)
+	msg.FastWriteNocopy(buf, nil)
+	return buf
+}
+
+// FastUnmarshal unmarshal the buf into msg. The msg should be generated by Kitex tool and implement ThriftMsgFastCodec.
+func FastUnmarshal(buf []byte, msg bthrift.ThriftMsgFastCodec) error {
+	_, err := msg.FastRead(buf)
+	return err
+}
+
+// for msgType of MarshalMsg
+// Please use theses consts instead of relying on apache thrift.TMessageType
+const (
+	CALL      = uint8(1)
+	REPLY     = uint8(2)
+	EXCEPTION = uint8(3)
+	ONEWAY    = uint8(4)
+)
+
+// MarshalMsg encodes the given msg to buf for generic thrift RPC.
+func MarshalMsg(method string, msgType uint8, seq int32, msg bthrift.ThriftMsgFastCodec) ([]byte, error) {
+	if method == "" {
+		return nil, errors.New("method not set")
+	}
+	sz := bthrift.Binary.MessageBeginLength(method, thrift.TMessageType(msgType), seq) + msg.BLength()
+	b := dirtmake.Bytes(sz, sz)
+	i := bthrift.Binary.WriteMessageBegin(b, method, thrift.TMessageType(msgType), seq)
+	_ = msg.FastWriteNocopy(b[i:], nil)
+	return b, nil
+}
+
+// UnmarshalMsg parses the given buf and stores the result to msg for generic thrift RPC.
+// for EXCEPTION msgType, it will returns `err` with *bthrift.ApplicationException type without storing the result to msg.
+func UnmarshalMsg(b []byte, msg bthrift.ThriftMsgFastCodec) (method string, seq int32, err error) {
```

Comment:
I think the old name ThriftMsgFastCodec will be confused with the Thrift message here. How about changing ThriftMsgFastCodec to ThriftFastCodec?